ui/v2.5/src/components/Settings/SettingsConfigurationPanel.tsx

```diff
@@ -82,6 +86,21 @@ export const SettingsConfigurationPanel: React.FC = () => {
     excludes,
     scraperUserAgent,
     scraperCDPPath,
+    stashBoxes: stashBoxes
+      .map(
+        (b) =>
+          ({
+            name: b?.name,
+            api_key: b?.api_key,
+            endpoint: b?.endpoint,
+          } as GQL.StashBoxInput)
+      )
+      .filter(
+        (b) =>
+          b.api_key &&
+          b.endpoint &&
+          (stashBoxes.length < 2 || (b?.name?.length ?? 0) > 0)
+      ),
```

Comment:
This validation should be done in the backend instead of or in addition to here. The server should not rely on the client to sanitise its inputs. If the configuration is invalid, the user should be notified as such.

With the current implementation, if a stash-box configuration is invalid it is removed altogether, rather than indicating to the user that it is invalid. Further, this also removed the first (valid) entry when the second entry was added and no names were entered.